tests/workflow_task_reported_problems_test.go

```diff
@@ -0,0 +1,360 @@
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/tests/testcore"
+)
+
+type WFTFailureReportedProblemsTestSuite struct {
+	testcore.FunctionalTestBase
+	tv                     *testvars.TestVars
+	initialRetryInterval   time.Duration
+	scheduleToCloseTimeout time.Duration
+	startToCloseTimeout    time.Duration
+
+	shouldFail atomic.Bool
+
+	activityRetryPolicy *temporal.RetryPolicy
+}
+
+func TestWFTFailureReportedProblemsTestSuite(t *testing.T) {
+	s := new(WFTFailureReportedProblemsTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) SetupTest() {
+	s.FunctionalTestBase.SetupTest()
+
+	s.tv = testvars.New(s.T()).WithTaskQueue(s.TaskQueue()).WithNamespaceName(s.Namespace())
+
+	s.initialRetryInterval = 1 * time.Second
+	s.scheduleToCloseTimeout = 30 * time.Minute
+	s.startToCloseTimeout = 15 * time.Minute
+
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+	}
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) simpleWorkflow(ctx workflow.Context) (string, error) {
+	if s.shouldFail.Load() {
+		panic("forced-panic-to-fail-wft")
+	}
+	return "done!", nil
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) simpleActivity() (string, error) {
+	return "done!", nil
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) workflowWithActivity(ctx workflow.Context) (string, error) {
+	var ret string
+	err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		ActivityID:             "activity-id",
+		DisableEagerExecution:  true,
+		StartToCloseTimeout:    s.startToCloseTimeout,
+		ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
+		RetryPolicy:            s.activityRetryPolicy,
+	}), s.simpleActivity).Get(ctx, &ret)
+	if err != nil {
+		return "", err
+	}
+
+	if s.shouldFail.Load() {
+		panic("forced-panic-to-fail-wft")
+	}
+
+	return "done!", nil
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) TestWFTFailureReportedProblems_SetAndClear() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	s.OverrideDynamicConfig(dynamicconfig.NumConsecutiveWorkflowTaskProblemsToTriggerSearchAttribute, 2)
+	s.shouldFail.Store(true)
+
+	s.Worker().RegisterWorkflow(s.simpleWorkflow)
+
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        testcore.RandomizeStr("wf_id-" + s.T().Name()),
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, s.simpleWorkflow)
+	s.NoError(err)
+
+	// Make sure the workflow has started and had an activity task scheduled and finished
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		require.NoError(t, err)
+		require.Equal(t, enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING, description.WorkflowExecutionInfo.Status)
+	}, 5*time.Second, 500*time.Millisecond)
```

Comment:
The fact that `ExecuteWorkflow` returned above indicates that the workflow is running. Delete this redundant code.